packages/flutter_tools/gradle/src/main/kotlin/dependency_version_checker.gradle.kts

```diff
@@ -235,7 +236,8 @@ class DependencyVersionChecker {
                         errorGradleVersion.toString(),
                         getPotentialGradleFix(project.getRootDir().getPath())
                     )
-                throw GradleException(errorMessage)
+                project.extra.set("failedDependencyChecks", true)
```

Comment:
So this may be to wild but can we return the exception as the value or we will run into the same namespacing issues.

Reply:
Return as in set it in the property value instead of `true`? I believe we would run in to the same issue unfortunately.